RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -0,0 +1,991 @@
+			}
+			break;
+		case $stat[Moxie]:
+			replace = equipped_item($slot[weapon]);
+			grab = $item[Backwoods Banjo];
+			if(can_equip($item[LOV Earrings]) && (item_amount($item[LOV Earrings]) > 0))
+			{
+				equip($slot[acc1], $item[LOV Earrings]);
+			}
+			break;
+		}
+
+		if((grab != $item[none]) && possessEquipment(grab) && (replace != grab) && can_equip(grab))
+		{
+			equip(grab);
+		}
+		if(!possessEquipment(grab) && (replace != grab) && (closet_amount(grab) > 0) && can_equip(grab))
+		{
+			closet = true;
+			take_closet(1, grab);
+			equip(grab);
+		}
+
+		visit_url("place.php?whichplace=chateau&action=chateau_restbox");
+
+		if((replace != grab) && (replace != $item[none]))
+		{
+			equip(replace);
+		}
+		cli_execute("outfit Backup");
+		if(closet)
+		{
+			if(item_amount(grab) > 0)
+			{
+				put_closet(1, grab);
+			}
+		}
+
+	}
+	else
+	{
+		set_property("restUsingChateau", false);
+		cli_execute("rest");
+		set_property("restUsingChateau", true);
+	}
+	return get_property("timesRested").to_int();
+}
+
+boolean haveFreeRestAvailable(){
+	return get_property("timesRested").to_int() < total_free_rests();
+}
+
+int freeRestsRemaining(){
+	return max(0, total_free_rests() - get_property("timesRested").to_int());
+}
+
+boolean haveAnyIotmAlternateCampsight(){
+	return chateaumantegna_available() || auto_campawayAvailable();
+}
+
+/*
+ * Try to use a free rest.
+ *
+ * returns true if a rest was used false if it wasnt (for any reason)
+ */
+boolean doFreeRest(){
+	if(haveFreeRestAvailable()){
+		int rest_count = get_property("timesRested").to_int();
+		return doRest() > rest_count;
+	}
+	return false;
+}
+
+float mp_regen()
+{
+	return 0.5 * (numeric_modifier("MP Regen Min") + numeric_modifier("MP Regen Max"));
+}
+
+float hp_regen()
+{
+	return 0.5 * (numeric_modifier("HP Regen Min") + numeric_modifier("HP Regen Max"));
+}
+
+boolean uneffect(effect toRemove)
+{
+	if(have_effect(toRemove) == 0)
+	{
+		return true;
+	}
+	if(($effects[Driving Intimidatingly, Driving Obnoxiously, Driving Observantly, Driving Quickly, Driving Recklessly, Driving Safely, Driving Stealthily, Driving Wastefully, Driving Waterproofly] contains toRemove) && (auto_get_campground() contains $item[Asdon Martin Keyfob]))
+	{
+		string temp = visit_url("campground.php?pwd=&preaction=undrive");
+		return true;
+	}
+
+	if(cli_execute("uneffect " + toRemove))
+	{
+		//Either we don\'t have the effect or it is shruggable.
+		return true;
+	}
+
+	if(item_amount($item[Soft Green Echo Eyedrop Antidote]) > 0)
+	{
+		visit_url("uneffect.php?pwd=&using=Yep.&whicheffect=" + to_int(toRemove));
+		print("Effect removed by Soft Green Echo Eyedrop Antidote.", "blue");
+		return true;
+	}
+	else if(item_amount($item[Ancient Cure-All]) > 0)
+	{
+		visit_url("uneffect.php?pwd=&using=Yep.&whicheffect=" + to_int(toRemove));
+		print("Effect removed by Ancient Cure-All.", "blue");
+		return true;
+	}
+	return false;
+}
+
+// Deprecated, please use acquireHP()
+boolean useCocoon()
```

Comment:
Maybe rename to `DEPRECATED_useCocoon()`? :)

Reply:
so id like to straight up remove it, but figure that will touch about every file increasing merge conflicts for this branch that is already pretty long lived. Once i merge this in I can sed it away and do a quick merge. This is just a stop gap.